linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala

```diff
@@ -32,7 +32,8 @@ import PolyfillableBuiltin._
 private[emitter] final class SJSGen(
     val jsGen: JSGen,
     val nameGen: NameGen,
```

Comment:
IIUC after this commit, we shouldn't use `NameGen` directly in most places.

I initially wanted to propose to make it private here, but I'm not sure this is feasible (just yet).

But maybe it makes sense to remove the wildcard imports in `CoreJSLib`, `FunctionEmitter` and `ClassEmitter` to make the remaining call-sites more explicit? (if not, it's IMO that big of a deal, because the test suite will catch bad / partial re-namings, now that we run all of it with the minifier).

Reply:
I did try that, but we still have a number of other, direct usages of `nameGen`. Notably for local names, label names, class captures, record field names. So at the moment it does not seem to be a win yet.